src/protocols/secure_channel/CheckinMessage.h

```diff
@@ -81,8 +92,19 @@ class DLL_EXPORT CheckinMessage
     static constexpr uint16_t sMinPayloadSize =
         CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES + sizeof(CounterType) + CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES;
 
-    // Issue #28603
-    static constexpr uint16_t sMaxAppDataSize = 1024;
+private:
+    /**
+     * @brief Generate the Nonce for the Check-In message
+     *
+     * @param[in]   hmacKeyHandle Key handle to use with the HMAC algorithm
+     * @param[in]   counter       Check-In Counter value to use as message of the HMAC algorithm
+     * @param[out]  output        output buffer for the generated Nonce.
+     *                            SUFFICIENT SPACE MUST BE ALLOCATED by the caller
+     *                            Size must be at least CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES
+     * @return CHIP_ERROR
```

Comment:
Please indicate the errors returned in all methods, for the common cases (e.g. CHIP_ERROR_BUFFER_TOO_SMALL versus CHIP_ERROR_INVALID_ARGUMENTS, etc).